src/main/java/sirius/web/security/UserContext.java

```diff
@@ -302,11 +302,17 @@ public void setCurrentUser(@Nullable UserInfo user) {
      */
     public void runAs(@Nullable UserInfo user, @Nonnull Runnable section) {
         UserInfo lastUser = getCurrentUser();
+        CallContext call = CallContext.getCurrent();
+        String lastLang = call.getLang();
         try {
             setCurrentUser(user);
```

Comment:
shouldn't we just set lang to null above and let the system figure out the rest (defaults etc.)